beacon-chain/blockchain/block_processing.go

```diff
@@ -104,6 +104,9 @@ func (c *ChainService) ReceiveBlock(ctx context.Context, block *pb.BeaconBlock)
 
 	beaconState, err = c.runStateTransition(headRoot, block, beaconState)
 	if err != nil {
+		if err := c.beaconDB.DeleteBlock(block); err != nil {
```

Comment:
This seems really dangerous to delete data if there is some arbitrary error.

What problem does this solve?